pkg/theiactl/commands/start.go

```diff
@@ -0,0 +1,378 @@
+	executorMemory      string
+}
+
+// startCmd represents the start command
+var startCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start a new policy recommendation Spark job",
+	Long: `Start a new policy recommendation Spark job.
+Network policies will be recommended based on the flow records sent by Flow Aggregator.
+Must finish the deployment of Theia first, please follow the steps in 
+https://github.com/antrea-io/theia/blob/main/docs/network-policy-recommendation.md`,
+	Example: `Start a policy recommendation spark job with default configuration
+$ theiactl policyreco start
+Start an initial policy recommendation spark job with network isolation option 1 and limit on last 10k flow records
+$ theiactl policyreco start --type initial --option 1 --limit 10000
+Start an initial policy recommendation spark job with network isolation option 1 and limit on flow records from 2022-01-01 00:00:00 to 2022-01-31 23:59:59.
+$ theiactl policyreco start --type initial --option 1 --start_time '2022-01-01 00:00:00' --end_time '2022-01-31 23:59:59'
+Start a policy recommendation spark job with default configuration but doesn't recommend toServices ANPs
+$ theiactl policyreco start --to_services=false
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var recoJobArgs []string
+		sparkResourceArgs := SparkResourceArgs{}
+
+		recoType, err := cmd.Flags().GetString("type")
+		if err != nil {
+			return err
+		}
+		if recoType != "initial" && recoType != "subsequent" {
+			return fmt.Errorf("recommendation type should be 'initial' or 'subsequent'")
+		}
+		recoJobArgs = append(recoJobArgs, "--type", recoType)
+
+		limit, err := cmd.Flags().GetInt("limit")
+		if err != nil {
+			return err
+		}
+		if limit < 0 {
+			return fmt.Errorf("limit should be an integer >= 0")
+		}
+		recoJobArgs = append(recoJobArgs, "--limit", strconv.Itoa(limit))
+
+		option, err := cmd.Flags().GetInt("option")
+		if err != nil {
+			return err
+		}
+		if option < 1 || option > 3 {
+			return fmt.Errorf("option of network isolation preference should be 1 or 2 or 3")
+		}
+		recoJobArgs = append(recoJobArgs, "--option", strconv.Itoa(option))
+
+		startTime, err := cmd.Flags().GetString("start_time")
+		if err != nil {
+			return err
+		}
+		if startTime != "" {
+			_, err = time.Parse("2006-01-02 15:04:05", startTime)
+			if err != nil {
+				return fmt.Errorf(`parsing start_time: %v, start_time should be in 
+'YYYY-MM-DD hh:mm:ss' format, for example: 2006-01-02 15:04:05`, err)
+			}
+			recoJobArgs = append(recoJobArgs, "--start_time", startTime)
+		}
+
+		endTime, err := cmd.Flags().GetString("end_time")
+		if err != nil {
+			return err
+		}
+		if endTime != "" {
+			_, err = time.Parse("2006-01-02 15:04:05", endTime)
+			if err != nil {
+				return fmt.Errorf(`parsing end_time: %v, end_time should be in 
+'YYYY-MM-DD hh:mm:ss' format, for example: 2006-01-02 15:04:05`, err)
+			}
+			recoJobArgs = append(recoJobArgs, "--end_time", endTime)
+		}
```

Comment:
Can we add a simple check to make sure that if both start and end time are specified start_time < end_time ?

Reply:
Sure, added